docs/kitematic/set_up_dev.md

```diff
@@ -0,0 +1,122 @@
+
+* <a href="http://electron.atom.io/" target="_blank"> Electron </a>
+
+* <a href="https://nodejs.org" target="_blank"> NodeJS </a>
+
+* <a href="https://facebook.github.io/react/" target="_blank"> React </a> and <a href="https://facebook.github.io/react/" target="_blank"> AltJS </a> which follows the Flux pattern
+
+## Download and install NVM
+
+To get started you will need to NodeJS v4.1.2. Using NVM (Node version manager) makes this easy.
+
+### Windows:
+1. Download  <a
+href="https://github.com/coreybutler/nvm-windows/releases/tag/1.1.0"
+target="_blank"> latest release </a>
+
+2. Follow the installer steps to get NVM installed. Please note, you need to
+uninstall any existing versions of node.js before installing NVM for Windows;
+the above installer link will have an uninstaller available.
+
+![windows installer](images/nvm_install.jpeg)
+
+### Mac OSX/Linux:
+
+1. Open a terminal window
+
+2. Copy and paste the following install script:
+
+  `curl -o-
+  https://raw.githubusercontent.com/creationix/nvm/v0.29.0/install.sh
+  | bash`
+
+  _More Info at the [Mac OSX/Linux official nvm
+  repo](https://github.com/creationix/nvm) or [Windows official nvm
+  repo](https://github.com/coreybutler/nvm-windows)_
+
+## Install NodeJS
+
+1. Install the proper NodeJS version and make it the default:
+* nvm install v4.1.2
+* nvm alias default v4.1.2
+
+2. Run `node --version` to confirm that you are now using the proper NodeJS
+version.
+
+## Fork the master branch of Kitematic
+
+To fork the master branch of Kitematic:
+
+1. Go to the <a href="https://github.com/kitematic/kitematic "target="_blank">
+kitematic/kitematic repository </a>.
+
+2. Click the "Fork" button in the upper right corner of the GitHub interface.
+
+    GitHub forks the repository to your GitHub account. The original
+    `kitematic/kitematic` repository becomes a new fork `YOUR_ACCOUNT/kitematic`
+    under your account.
+
+3. Copy your fork's clone URL from GitHub.
+
+    GitHub allows you to use HTTPS or SSH protocols for clones. You can use the `git` command line or clients like Subversion to clone a repository. This guide assume you are using the HTTPS protocol and the `git` command line. If you are comfortable with SSH and some other tool, feel free to use that instead. You'll need to convert what you see in the guide to what is appropriate to your tool.
+
+## Clone your repository locally and create a branch
+
+To clone your repository and create a branch for the issue:
+
+1. Open a terminal window on your local host and change to your home directory.
+
+    $ cd ~
+
+    In Windows, you'll work in your Docker Quickstart Terminal window instead of Powershell or a `cmd` window.
+
+2. Clone the fork to your local host into a repository.
+
+    `$ git clone https://github.com/YOUR_USERNAME/kitematic.git`
+
+3. Create and checkout a branch for the issue you will be working on
+
+    `$ git checkout -b XXXX-branch` (where XXX is the number of the issue previously noted)
+
```

Comment:
We should include a step to set an upstream remote that points to the original source repo so that they can track changes and re-base their local repo to it. We can either have them set this up here, or when they are doing the pull request. We are linking out to a topic that covers this (link it currently broken--need to fix, URL has changed), but it might be worth pulling this detail into our procedure if even in short form. See http://docs.docker.com/opensource/project/set-up-git/ and see "Set your signature and an upstream remote".